crates/ra_assists/src/assists/invert_if.rs

```diff
@@ -0,0 +1,89 @@
+use hir::db::HirDatabase;
+use ra_syntax::ast::{self, AstNode};
+use ra_syntax::{TextRange, TextUnit};
+
+use super::apply_demorgan::undo_negation;
+use crate::{Assist, AssistCtx, AssistId};
+
+// Assist: invert_if
+//
+// Apply invert_if
+// This transforms if expressions of the form `if !x {A} else {B}` into `if x {B} else {A}`
+// This also works with `!=`. This assist can only be applied with the cursor
+// on `if`.
+//
+// ```
+// fn main() {
+//     if<|> !y {A} else {B}
+// }
+// ```
+// ->
+// ```
+// fn main() {
+//     if y {B} else {A}
+// }
+// ```
+
+pub(crate) fn invert_if(ctx: AssistCtx<impl HirDatabase>) -> Option<Assist> {
+    let expr = ctx.find_node_at_offset::<ast::IfExpr>()?;
+    let expr_range = expr.syntax().text_range();
+    let if_range = TextRange::offset_len(expr_range.start(), TextUnit::from_usize(2));
+    let cursor_in_range = ctx.frange.range.is_subrange(&if_range);
```

Comment:
If we are looking specifically for `if` token, it might be better to start with `ctx.find_token_at_offset(T![if])`, and cast it's parent to `IfExpr`